src/measure_theory/lp_space.lean

```diff
@@ -588,6 +588,20 @@ begin
   exact snorm'_add_le hf hg hp1_real,
 end
 
+lemma snorm'_sum_le [second_countable_topology E] {ι} {f : ι → α → E} {s : finset ι}
+  (hfs : ∀ i, i ∈ s → ae_measurable (f i) μ) (hq1 : 1 ≤ q) :
+  snorm' (∑ i in s, f i) q μ ≤ ∑ i in s, snorm' (f i) q μ :=
+@finset.le_sum_of_subadditive_on_pred (α → E) ℝ≥0∞ ι _ _ (λ f, snorm' f q μ)
+  (λ f, ae_measurable f μ) (snorm'_zero (zero_lt_one.trans_le hq1))
+  (λ f g hf hg, snorm'_add_le hf hg hq1) (λ x y, ae_measurable.add) _ _ hfs
```

Comment:
You have put enough explicit arguments in your lemma that you don't need the @ version here:
```suggestion
finset.le_sum_of_subadditive_on_pred (λ (f : α → E), snorm' f q μ)
  (λ f, ae_measurable f μ) (snorm'_zero (zero_lt_one.trans_le hq1))
  (λ f g hf hg, snorm'_add_le hf hg hq1) (λ x y, ae_measurable.add) _ hfs
```

Reply:
Thanks, this is great! I tried removing the @ several times without success. But I did not add the type of `f` in `(λ (f : α → E), snorm' f q μ)` as you did.

Reply:
It wasn't magic, I just tried to think as if I were Lean. When Lean sees `(λ f, snorm' f q μ)`, it knows that the target space has to be `ℝ≥0∞` (as this is where `snorm'` takes its values), but it has no way to know the source space because `snorm'` is defined on many spaces. So I just added the information it could not guess, and then everything worked fine.